fe/fe-core/src/main/java/org/apache/doris/nereids/util/SlotExtractor.java

```diff
@@ -49,24 +47,24 @@ public static Set<Slot> extractSlot(Collection<Expression> expressions) {
      * extract slot reference.
      */
     public static Set<Slot> extractSlot(Expression... expressions) {
-
-        Set<Slot> slots = Sets.newLinkedHashSet();
+        Set<Slot> slots = Sets.newHashSet();
         for (Expression expression : expressions) {
             slots.addAll(extractSlot(expression));
         }
         return slots;
     }
 
-    private static List<Slot> extractSlot(Expression expression) {
-        List<Slot> slots = Lists.newArrayList();
-        new SlotExtractor().visit(expression, slots);
+    private static Set<Slot> extractSlot(Expression expression) {
```

Comment:
if this function is private, i think `void extractSlot(Expression expression, Set<Slot>)` maybe better